blueflood-integration-tests/pom.xml

```diff
@@ -0,0 +1,199 @@
+              <goal>prepare-agent</goal>
+            </goals>
+          </execution>
+          <execution>
+            <id>prepare-it-agent</id>
+            <phase>pre-integration-test</phase>
+            <goals>
+              <goal>prepare-agent-integration</goal>
+            </goals>
+            <configuration>
+              <propertyName>jacoco.agent.it.argLine</propertyName>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-failsafe-plugin</artifactId>
+        <version>2.19.1</version>
+        <configuration>
+          <systemPropertyVariables>
+            <!-- Load properties for the IT's from the module-local properties file -->
+            <blueflood.config>file:${basedir}/src/integration-test/resources/blueflood.properties</blueflood.config>
+          </systemPropertyVariables>
+          <!--
+          Include the jacoco agent for maven builds; note that IntelliJ WILL NOT resolve this property and won't apply
+          the arg line as a result. Don't put anything else important here! Use the IT blueflood.properties or some
+          other config mechanism appropriate to the integration tests.
+          -->
+          <argLine>${jacoco.agent.it.argLine}</argLine>
+        </configuration>
+        <executions>
+          <!-- States that both integration-test and verify goals of the Failsafe Maven plugin are executed. -->
+          <execution>
+            <id>integration-tests</id>
+            <goals>
+              <goal>integration-test</goal>
+            </goals>
+            <configuration>
+              <systemPropertyVariables>
+                <CASSANDRA_HOSTS>${cassandra.listenAddress}:${cassandra.rpcPort}</CASSANDRA_HOSTS>
+                <DEFAULT_CASSANDRA_PORT>${cassandra.rpcPort}</DEFAULT_CASSANDRA_PORT>
+                <CASSANDRA_BINXPORT_PORT>${cassandra.nativeTransportPort}</CASSANDRA_BINXPORT_PORT>
+                <CASSANDRA_BINXPORT_HOSTS>${cassandra.listenAddress}:${cassandra.nativeTransportPort}</CASSANDRA_BINXPORT_HOSTS>
+                <CASSANDRA_DRIVER>datastax</CASSANDRA_DRIVER>
+              </systemPropertyVariables>
+              <testSourceDirectory>src/integration-test/java</testSourceDirectory>
+              <!-- Skips integration tests if the value of skip.integration.tests property is true -->
+              <skipTests>${skip.integration.tests}</skipTests>
+              <includes>
+                <include>**/*Integration*.java</include>
+              </includes>
+              <excludes>
+                <!-- requires an API service -->
+                <exclude>**/InternalApiIntegrationTest.java</exclude>
+                <!-- this one requires a live zookeeper -->
+                <exclude>**/ZKShardLockManagerIntegrationTest.java</exclude>
+                <exclude>**/ScheduleContextIntegrationTest.java</exclude>
```

Comment:
We need to remember to take a look at these tests. If they're always turned off, what's the point of even having them?

Reply:
noted